### DISCUSSION TITLE
Why do I get a package type 'DotnetTool' error when migrating from Wix 3.11.2 to 4.0.4?

### DISCUSSION THREAD

**Discussion by SCLD-JBuczkowski** [wixtoolset/issues#8014] [Questions]
I am a total newbie. I inherited a .Net 4.8 Framework solution that uses WiX to create installer packages.
The solution currently uses WiX 3.11.2 (which is deprecated) and I'm trying to upgrade to the latest version (4.04)
I've watched some of the Dojo videos to come up to speed with WiX and migrating 3.x to 4.x.
I plan on upgrading the Wixproj files through VS (Upgrade to Wix 4...)

In examining the .Net projects, I notice that they include references to the WiX 3.11.2 Nuget package.
In order to migrate to 4.x, I assume that I have to update these project Nuget references as well.

When I attempt to update via Nuget package manager, I get the following error message:
Package 'wix 4.0.4' has a package type 'DotnetTool' that is not supported by project.

I've googled around, but am at a loss as to why simply upgrading the package version would cause this error.
What is going on here, and what steps do I need to take to resolve this?

Thanks,
JohnB

**Comment by robmen** [comment]
Try using [HeatWave to upgrade your project](https://www.firegiant.com/docs/heatwave/convert/) and look at the before and after to see the differences.

You'll also want to watch the [Deployment Dojo S1:E1 - Introducing WiX v4 to the Deployment Dojo](https://robmensching.com/deployment-dojo/episodes/s1/e1/introducing-wix-v4-to-the-deployment-dojo/) and then [Deployment Dojo S1:E8 - A Perfect Match: WiX and MSBuild](https://robmensching.com/deployment-dojo/episodes/s1/e8/a-perfect-match-wix-and-msbuild/) closely to see the difference between using `wix.exe` from the command-line and `WixToolset.Sdk` from MSBuild with a .wixproj.

If you want guaranteed answers while learning software installation, check out [FireGiant Enterprise support](https://www.firegiant.com/services/support-programs/).

**Reply by SCLD-JBuczkowski** [reply]
I did use Heatwave to upgrade the project.  The results I reported were after the HeatWave "Auto-Upgrade".

Once again.., why am I getting this error when upgrading the source code package references to 4.04?
    **When I attempt to update via Nuget package manager, I get the following error message:
    Package 'wix 4.0.4' has a package type 'DotnetTool' that is not supported by project.**

Has the build/project type somehow changed between releases for the DLL in that NuGet Pkg?

**Reply by robmen** [reply]
> Has the build/project type somehow changed between releases for the DLL in that NuGet Pkg?

Yes. Radically. The [intro documentation](https://wixtoolset.org/docs/intro/) explains it.
